File: Python/Line Detection.py
import cv2
import numpy as np
import matplotlib.pyplot as plt
img = cv2.imread(r'Enter your image path here')
img_rgb = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
edges_r = cv2.Canny(img[:,:,2], 100, 200)
edges_g = cv2.Canny(img[:,:,1], 100, 200)  
edges_b = cv2.Canny(img[:,:,0], 100, 200)  
edges = edges_r | edges_g | edges_b
lines = cv2.HoughLinesP(edges, rho=1, theta=np.pi/180, threshold=100, 
                        minLineLength=7, maxLineGap=5)
result = img_rgb.copy()
if lines is not None:
    for line in lines:
        x1, y1, x2, y2 = line[0]
        cv2.line(result, (x1, y1), (x2, y2), (0, 255, 0), 2)
        cv2.drawMarker(result, (x1, y1), (255, 255, 0), markerType=cv2.MARKER_CROSS, 
                      markerSize=10, thickness=2)
        cv2.drawMarker(result, (x2, y2), (255, 0, 0), markerType=cv2.MARKER_CROSS, 
                      markerSize=10, thickness=2)
plt.figure(figsize=(10, 8))
plt.imshow(result)
plt.title('Detected Lines')
plt.axis('off')
plt.show()

File: MATLAB/Line Detection.m
img = imread('Enter your image path here');
edgesR = edge(img(:, :, 1), 'Canny');
edgesG = edge(img(:, :, 2), 'Canny');
edgesB = edge(img(:, :, 3), 'Canny');
edges = edgesR | edgesG | edgesB;
[H, T, R] = hough(edges);
P = houghpeaks(H, 5, 'threshold', ceil(0.3 * max(H(:))));
lines = houghlines(edges, T, R, P, 'FillGap', 5, 'MinLength', 7);
figure;
imshow(img), hold on;
for k = 1:length(lines)
    xy = [lines(k).point1; lines(k).point2];
    plot(xy(:,1), xy(:,2), 'LineWidth', 2, 'Color', 'green');
    plot(xy(1,1), xy(1,2), 'x', 'LineWidth', 2, 'Color', 'yellow');
    plot(xy(2,1), xy(2,2), 'x', 'LineWidth', 2, 'Color', 'red');
end
title('Detected Lines');
hold off;
